enterprise/server/remote_execution/block_io/block_io_unix.go

```diff
@@ -0,0 +1,107 @@
+//go:build unix
+
+package block_io
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+// Device represents a block device.
+type Device struct {
+	// Major and Minor device numbers.
+	Maj, Min int64
+}
+
+// LookupBlockDevice returns the block device for the filesystem mounted at
+// the given path.
+func LookupDevice(path string) (*Device, error) {
+	// Get the device like /dev/sda1 etc.
+	dev, err := getDeviceFromPath(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// Stat the device to get Rdev, then extract major/minor device nums.
+	fileInfo, err := os.Stat(dev)
+	if err != nil {
+		return nil, fmt.Errorf("stat %q: %w", dev, err)
+	}
+	stat, ok := fileInfo.Sys().(*syscall.Stat_t)
+	if !ok {
+		return nil, err
+	}
+	major := unix.Major(stat.Rdev)
+	minor := unix.Minor(stat.Rdev)
+	blkDev := &Device{
+		Maj: int64(major),
+		Min: int64(minor),
+	}
+
+	return blkDev, nil
+}
+
+// ParseMajMin parses major/minor block device numbers formatted like
+// "MAJ:MIN"
+func ParseMajMin(str string) (*Device, error) {
```

Comment:
This should probably return two ints instead of Device. There's no guarantee that the string represents a real device.

Reply:
good point - done